pkg/merkletree/tree.go

```diff
@@ -0,0 +1,173 @@
+package merkletree
+
+import (
+	"bytes"
+	"errors"
+	"sort"
+
+	"github.com/ethereum/go-ethereum/crypto"
+)
+
+// DefaultHashFunc is the default hash function in case none is passed.
+var DefaultHashFunc = crypto.Keccak256
```

Comment:
const is better

Reply:
That comment was more a not for myself 😅

It's intentional because clients might need to know what hash was used to generate the leaves, so they can use the same. Ideally, that value would be a const, because you don't want clients messing with that. I'll probably switch to `hash.Hash`.

Reply:
Sounds good, but I think its fine to think about `pkg/merkletree` just as an algorithm implementation.
If `Keccak256` is relevant for a higher-level protocol, that's fine but can be considered relevant in an outer context.

But... biasing this package towards the protocol sounds fine too, it can help people importing `go-tableland` to use this package directly without "any further" consideration.